heka/_service.sls

```diff
@@ -0,0 +1,209 @@
+{%- macro service_config(service_name) %}
+
+{%- set server = salt['pillar.get']('heka:'+service_name) %}
+
+{%- if server.enabled %}
+
+heka_{{ service_name }}_log_file:
+  file.managed:
+  - name: /var/log/{{ service_name }}.log
+  - user: heka
+  - mode: 750
+  - replace: False
+
+heka_{{ service_name }}_conf_dir:
+  file.directory:
+  - name: /etc/{{ service_name }}
+  - user: root
+  - mode: 750
+  - makedirs: true
+
+heka_{{ service_name }}_conf_dir_clean:
+  file.directory:
+  - name: /etc/{{ service_name }}
+  - clean: true
+
+{%- if grains.get('init', None) == 'systemd' %}
+
+heka_{{ service_name }}_service_file:
+  file.managed:
+  - name: /etc/init/{{ service_name }}.conf
+  - user: root
+  - mode: 644
+  - template: jinja
+
+{%- else %}
+
+heka_{{ service_name }}_service_file:
+  file.managed:
+  - name: /etc/init/{{ service_name }}.conf
+  - user: root
+  - mode: 644
+  - template: jinja
```

Comment:
In this declaration exactly the same as the previous one? If that's the case it can probably be moved outside the if/else block.